erts/emulator/beam/atom.c

```diff
@@ -105,19 +105,28 @@ more_atom_space(void)
  * Allocate string space within an atom text segment.
  */
 
-static byte*
-atom_text_alloc(int bytes)
+static Eterm*
+atom_text_alloc(int bytes, Uint* size)
 {
     byte *res;
 
     ASSERT(bytes <= MAX_ATOM_SZ_LIMIT);
+    if (bytes <= ERL_ONHEAP_BINARY_LIMIT) {
+        bytes = heap_bits_size(NBITS(bytes));
+    } else {
+        bytes = ERL_REFC_BITS_SIZE;
+    }
+    bytes *= sizeof(Eterm);
```

Comment:
This looks like you're allocating an entire Eterm per byte, which (a) is wasteful, and (b) I can't see the need for.

Reply:
`atom_text_alloc` is a bit of a misnomer with this PR: it allocates space for the literal binary term, not for the text as such. The PR also got merged by accident and will be reworked later on :)

Reply:
Yes, but even in a binary we shouldn't need an entire Eterm (8 bytes on most current HW) per byte in the text. Oh well, maybe I'm confused by the unfinished state of this PR. Looking forward to review the finished one.